kahuna/public/js/components/gr-sort-control/gr-sort-control.tsx

```diff
@@ -0,0 +1,182 @@
+import * as React from "react";
+import * as angular from "angular";
+import { react2angular } from "react2angular";
+import { useEffect, useState, KeyboardEvent } from "react";
+
+import "./gr-sort-control.css";
+
+const SELECT_OPTION = "Select an option";
+const DEFAULT_OPTION = "uploadNewOld";
+const COLLECTION_OPTION = "collecAdded";
+const CONTROL_TITLE = "Sort by:";
+
+const downArrowIcon = () =>
+  <svg width="12" height="12" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
+    <path d="M11.178 19.569a.998.998 0 0 0 1.644 0l9-13A.999.999 0 0 0 21 5H3a1.002 1.002 0 0 0-.822 1.569l9 13z"/>
+  </svg>;
+
+const emptyIcon = () =>
+  <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
+    <rect width="100%" height="100%" fill="none" stroke="none" />
+  </svg>;
+
+const tickIcon = () =>
+  <svg width="24" height="24" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
+    <polyline fill="none" stroke="inherit" points="3.7 14.3 9.6 19 20.3 5" strokeLinecap="round" strokeLinejoin="round" strokeWidth="2"/>
+  </svg>;
+
+export interface SortDropdownOption {
+  value: string;
+  label: string;
+}
+
+export interface SortDropdownProps {
+  options: SortDropdownOption[];
+  selectedOption?: SortDropdownOption | null;
+  onSelect: (option: SortDropdownOption) => void;
+  query?: string | "";
+  orderBy?: string | "";
+}
+
+export interface SortWrapperProps {
+  props: SortDropdownProps;
+}
+
+const SortControl: React.FC<SortWrapperProps> = ({ props }) => {
+  const defOptVal:string = DEFAULT_OPTION;
+  const [hasCollection, setHasCollection] = useState(false);
+  const options = props.options;
+  const defSort:SortDropdownOption = options.filter(opt => opt.value == defOptVal)[0];
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedOption, setSelection] = useState(defSort);
+  const [currentIndex, setCurrentIndex] = useState(-1);
+
+  const checkForCollection = (query:string): boolean => /~"[a-zA-Z0-9 #-_.://]+"/.test(query);
+
+  const hasClassInSelfOrParent = (node: Element | null, className: string): boolean => {
+    if (node !== null && node.classList.contains(className)) {
+      return true;
+    }
+
+    while (node && node.parentNode && node.parentNode !== document) {
+      node = node.parentNode as Element;
+      if (node.classList.contains(className)) {
+        return true;
+      }
+    }
+
+    return false;
+  };
+
+  const autoHideListener = (event: any) => {
```

Comment:
from github actions: unexpected any, specify a different type

Reply:
understood and accepted